### ISSUE TITLE
All inlay hints are at the end of the line

### ISSUE THREAD

**Issue by marekdedic** [neoclide/coc.nvim#4234]
## Result from CocInfo

```
## versions

vim version: NVIM v0.7.2
node version: v18.9.1
coc.nvim version: 0.0.82-043b4310 2022-09-24 15:40:38 +0800
coc.nvim directory: /home/marekdedic/.local/share/nvim/plugged/coc.nvim
term: tmux
platform: linux

## Log of coc.nvim

2022-09-25T09:41:42.344 INFO (pid:192563) [services] - registered service "tsserver"
2022-09-25T09:41:42.346 INFO (pid:192563) [plugin] - coc.nvim initialized with node: v18.9.1 after 118ms
2022-09-25T09:41:42.374 INFO (pid:192563) [services] - service tsserver started
2022-09-25T09:41:42.727 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:41:47.443 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:41:48.737 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:41:49.854 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:41:51.178 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:41:52.591 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:42:21.425 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:42:22.218 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:42:22.758 WARN (pid:192563) [completion] - Suggest not triggered with input "", minimal trigger input length: 1
2022-09-25T09:42:24.913 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:42:26.796 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:42:27.497 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:43:00.110 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:43:03.097 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:43:03.807 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:43:07.987 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:43:13.166 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:43:16.328 INFO (pid:192563) [attach] - receive notification: highlight []
2022-09-25T09:44:29.929 INFO (pid:192563) [attach] - receive notification: showInfo []
```

## Describe the bug

When I enabled inlay hints in coc-tsserver, all of the inlays are placed at the end of the line.

## Reproduce the bug

**We will close your issue when you don't provide minimal vimrc and we can't
reproduce it**

- Create file `mini.lua` with：

  ```lua
  local Plug = vim.fn['plug#']

  vim.call('plug#begin', vim.fn.stdpath('data') .. '/plugged')

  Plug('neoclide/coc.nvim', {branch = 'release'})
  Plug('neoclide/coc-tsserver', {['do'] = 'yarn install --frozen-lockfile'})

  vim.call('plug#end')
  ```

- Start (neo)vim with command: `vim -u mini.lua`

- Operate vim.

## Screenshots (optional)

![image](https://user-images.githubusercontent.com/3134692/192133491-e1495bd0-5d7b-44bc-9813-16750e26dce8.png)

**Comment by iamcco** [comment]
It need neovim to support display virtual text at middle line.

**Comment by chemzqm** [comment]
No possible unitl https://github.com/neovim/neovim/pull/9496

**Comment by marekdedic** [comment]
Oh, ok, I thought that was already implemented as part of virttext, thanks for pointing me to the relevant issue :)

**Comment by oblitum** [comment]
NeoVim folks finished borrowing Vim's inline virtual text: https://github.com/neovim/neovim/pull/20130.

With this, I think coc.nvim can already take advantage of it, just like it already does for Vim, in NeoVim master atm. I'm not sure whether it's supposed to worked out of the box without any change, I'm not on NeoVim master atm and didn't try whether it's already working or not.

Worth to mention that their own bundled LSP server will start taking advantage of this Vim feature once https://github.com/neovim/neovim/pull/23736 is merged, but coc doesn't need to wait for that to happen.

**Comment by oblitum** [comment]
I think it's now valid reopening this issue and close it once a commit comes that at least documents coc supports in on NeoVim, just as in Vim?

**Comment by fannheyward** [comment]
@oblitum https://github.com/neoclide/coc.nvim/pull/4648

**Comment by oblitum** [comment]
Hi @fannheyward. I was checking out our profiles on activities and I've noticed @chemzqm is out _focusing_ atm focusing on other matters, so I guess it's on us to commit/approve?

I'll try your PR asap, though I'm a bit busy with life matters too, but I'm gonna do it in a week or so.

**Comment by oblitum** [comment]
If you force push that PR commenting "Closes #4234", than we can let this issue open and it'll get closed once that is merged.

**Comment by oblitum** [comment]
I see you have other PRs on hold, and I'm unsure whether you're blocked on lack of review/approve, if so, I might give them them a look too and approve if it's OK, and harmless.

**Comment by fannheyward** [comment]
> let this issue open and it'll get closed once that is merged.

I don't think it's needed

> blocked on lack of review/approve

Yes.

**Comment by oblitum** [comment]
OK.

**Comment by chaozwn** [comment]
@fannheyward @chemzqm , excuse me, now neovim: https://github.com/neovim/neovim/pull/23426 has been merged. i think we can reopen this issue. and make inlay hints behind parameter or function, not all inlay hints show at the end of the line.
